// directus/dist/database/helpers/fn/dialects/oracle.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FnHelperOracle = void 0;
const types_1 = require("../types");
class FnHelperOracle extends types_1.FnHelper {
    year(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'IYYY')", [table, column]);
    }
    month(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'MM')", [table, column]);
    }
    week(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'IW')", [table, column]);
    }
    day(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'DD')", [table, column]);
    }
    weekday(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'D')", [table, column]);
    }
    hour(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'HH24')", [table, column]);
    }
    minute(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'MI')", [table, column]);
    }
    second(table, column) {
        return this.knex.raw("TO_CHAR(??.??, 'SS')", [table, column]);
    }
    count(table, column) {
        var _a, _b, _c, _d, _e;
        const type = (_e = (_d = (_c = (_b = (_a = this.schema.collections) === null || _a === void 0 ? void 0 : _a[table]) === null || _b === void 0 ? void 0 : _b.fields) === null || _c === void 0 ? void 0 : _c[column]) === null || _d === void 0 ? void 0 : _d.type) !== null && _e !== void 0 ? _e : 'unknown';
        if (type === 'json') {
            return this.knex.raw("json_value(??.??, '$.size()')", [table, column]);
        }
        if (type === 'alias') {
            return this._relationalCount(table, column);
        }
        throw new Error(`Couldn't extract type from ${table}.${column}`);
    }
}
exports.FnHelperOracle = FnHelperOracle;
